src/lib.rs

```diff
@@ -214,6 +214,26 @@ impl From<DiversifierIndex> for u128 {
     }
 }
 
+impl PartialOrd for DiversifierIndex {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for DiversifierIndex {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.0
+            .iter()
+            .rev()
+            .zip(other.0.iter().rev())
+            .find_map(|(a, b)| match a.cmp(b) {
```

Comment:
Doesn't a `[u8]` already have an ordering? I.e. (untested) it should be possible to compare the reversed arrays directly.